contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3Propagator.java

```diff
@@ -62,8 +62,13 @@ public B3Propagator() {
    * @param singleHeader whether to use single or multiple headers.
    */
   public B3Propagator(boolean singleHeader) {
```

Comment:
use a `create` factory method instead of a public ctor.

Reply:
Talking of, since the `Propagators` do not keep state, I'm wondering on why we don't expose them as static instances, i.e. `getSingleHeaderPropagator()` and `getMultipleHeaderPropagator()`.

Reply:
@bogdandrutu  I can make this change. But not sure why we would need a factor when we just have one argument.

@carlosalberto  Now I think about this, making static methods make sense. But, again I don't know what the implication will be. I think this will cause a breaking change for the clients while upgrading to the next version.

Reply:
Breaking changes are ok at the moment, since we haven't hit version 1.0 yet.

Reply:
Yeah, it seems like a `B3Propagators.SINGLE_HEADER` and `B3Propagators.MULTI_HEADER` would be a great way to get access to them.

Reply:
Factory methods are better than constructors because they can be change to return different instances for example.

Reply:
Happy with @jkwatson solution fyi

Reply:
I have made changes based on the feedback.

I have made the current constructor as `private`. I have exposed a public static variable for `B3Propagators.SINGLE_HEADER` and `B3Propagators.MULTIPLE_HEADER`.

Reply:
One last thing: do we want to expose these singletons directly, or should we instead expose them through a static methods? We have historically chosen the second option, which I think it's the slightly better approach ;)

Reply:
no strong opinion from me. For consistency, it's probably better to hide the instances behind static accessors.

Reply:
Made change to expose the singletons via static getters.